src/jdk.jfr/share/classes/jdk/jfr/internal/ChunkInputStream.java

```diff
@@ -62,13 +64,18 @@ public int available() throws IOException {
     }
 
     private boolean nextStream() throws IOException {
-        if (!nextChunk()) {
-            return false;
+        while (nextChunk()) {
+            try {
+                stream = new BufferedInputStream(Files.newInputStream(currentChunk.getFile()));
```

Comment:
Should we close `stream` if it's not null?

Reply:
If `nextChunk` returns `true` e.g. 2 times in a row, the second iteration of the while loop at L67 will overwrite `stream` without closing the previous instance. I don't see if/where `closeStream` will be called within two iterations of the loop, maybe I'm missing something?

Reply:
Thanks for the clarification